databroker/broker/struct.py

```diff
@@ -19,6 +20,26 @@ def __init__(self, mongo_document):
             if isinstance(attr, mongoengine.Document):
                 attr = BrokerStruct(attr)
             setattr(self, field, attr)
+        try:
```

Comment:
This whole thing is moving into metadatastore, where it will be imported for use by the broker in building channelarchiver events. Mind if I copy/paste?